tensorflow/core/api_def/base_api/api_def_GenerateBoundingBoxProposals.pbtxt

```diff
@@ -0,0 +1,81 @@
+op {
+  graph_op_name: "GenerateBoundingBoxProposals"
+  visibility: HIDDEN
+  in_arg {
+    name: "scores"
+    description: <<END
+A 4-D float tensor of shape `[num_images, height, width, num_achors]` containing scores of the boxes for given anchors, can be unsorted.
+END
+  }
+  in_arg {
+    name: "bbox_deltas"
+    description: <<END
+A 4-D float tensor of shape `[num_images, height, width, 4 x num_anchors]`. encoding boxes with respec to each anchor.
+Coordinates are given in the form [dy, dx, dh, dw].
+END
+  }
+  in_arg {
+    name: "image_info"
+    description: <<END
+A 2-D float tensor of shape `[num_images, 5]` containing image information Height, Width, Scale.
```

Comment:
why there are 5 elements each row if it only has height, width and scale? This makes no sense

Reply:
Yeah, [generate_box_proposals_op.cu.cc#L115~118](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/image/generate_box_proposals_op.cu.cc#L115) contains the following...
```C
const float img_height = d_img_info_vec[5 * image_index + 0];
const float img_width = d_img_info_vec[5 * image_index + 1];
const float min_size_scaled = min_size * d_img_info_vec[5 * image_index + 2];
```
...at which point one would expect `5 * image_index + 3` to be assigned to some `foo` and a `5 * image_index + 4` to some `bar`, but this never happens. These two float elements effectively ending up being nothing more than some weird unused padding, it seems.

I suspect they were used as extra attributes at some earlier development stage, and code wasn't properly updated upon their removal?

@aaroey @pengchongjin 
Any reason why this didn't seem like cause for concern during (pre-merge) review?

Reply:
Actually, I guess the padding might be intended as a bank conflict avoidance tactic. Either way, implementation details shouldn't distort the public API; and if they do anyway there should be some comment/documentation mentioning the underlying reasoning.